src/main/java/com/github/manolo8/darkbot/gui/MapDrawer.java

```diff
@@ -187,26 +209,63 @@ public MapGraphicsImpl(StarSystemAPI star, ConfigAPI config) {
             this.cs = config.requireConfig("bot_settings.map_display.cs");
             this.fonts = config.requireConfig("bot_settings.map_display.cs.fonts");
             this.displayFlags = config.requireConfig("bot_settings.map_display.toggle");
+            this.radiationScale = config.requireConfig("bot_settings.map_display.radiation_scale");
         }
 
-        public void setup(Graphics graphics, int width, int height) {
+        public void setup(Graphics g2, int width, int height) {
             this.width = width;
             this.height = height;
             this.widthMid = width / 2;
             this.heightMid = height / 2;
 
-            this.g2 = (Graphics2D) graphics; //graphics.create();
-
-            this.setColor("background");
-            this.g2.fillRect(0, 0, getWidth(), getHeight());
-            this.g2.setRenderingHints(RENDERING_HINTS);
+            this.scaleX = mapBounds.getWidth() / width;
+            this.scaleY = mapBounds.getHeight() / height;
+
+            this.g2 = (Graphics2D) g2;
+            this.g2.addRenderingHints(RENDERING_HINTS);
+
+            double scale = getRadScale();
+            if (scale > 0) {
+                x = (int) ((mapBounds.getWidth() / 2 * scale) / (mapBounds.getWidth() / width));
+                y = (int) ((mapBounds.getHeight() / 2 * scale) / (mapBounds.getHeight() / height));
```

Comment:
I *think* these can be simplified to `scale * width / 2`, map bounds being both above and below the division makes them redundant terms that can be removed